### PR TITLE
gensym vars now use X__gensymY instead of X`gensymY

### PR DESCRIPTION
**Pull request by ringabout**
close https://github.com/nim-lang/Nim/pull/17304

### CODE REVIEW

**Comment by juancarlospaco** [comment]
Why not `_generated_symbol` instead of `gensym` ?. 🤔

**Comment by Araq** [comment]
> Why not _generated_symbol instead of gensym ?

Because it affects the size of the generated C code and compile-times are never fast enough.

**Comment by ringabout** [comment]
It seems that https://github.com/iffy/nim-argparse is broken because it uses the internal implementation of `gensym`:

```nim
  if "`gensym" in node.strVal:
    return ident(node.strVal.split("`")[0])
```

ref https://github.com/iffy/nim-argparse/blob/4048a76b24919807b0871dd145ba93a48c2688e7/src/argparse/macrohelp.nim#L35

**Comment by ringabout** [comment]
Should I use `optNimV16Emulation` or define `nimOldGenSym` to keep backward compatibility?

**Comment by Araq** [comment]
Looking at the patch I don't like the solution anymore. We need to find a better way.

File: compiler/astalgo.nim [hunk 1]
```diff
@@ -194,8 +194,18 @@ proc getSymFromList*(list: PNode, ident: PIdent, start: int = 0): PSym =
...
+    if a[alen] == '_':
+      if a[alen-1] == '_':
+        dec alen
+        isGenSym = true
+        break
+      else:
+        dec(alen, 2)
+    else:
+      dec alen
+  if not isGenSym: alen = a.len
```
Comment (ringabout) [inline compiler/astalgo.nim:211]: something like this should work too
```nim
  let alen = rfind(a, "__")
  if alen < 0:
    alen = a.len
```

**Pull request closed without merging**